### PR TITLE
Model serving test suite

### PR DESCRIPTION
**Pull request by lugi0**
Signed-off-by: Luca Giorgi <lgiorgi@redhat.com>

### CODE REVIEW

File: tests/Resources/Files/uninstall-model-serving.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#! /bin/bash
+
+MODELMESH_PROJECT=${1:-"redhat-ods-applications"}
+INFERENCE_SERVICE_PROJECT=${2:-"mesh-test"}
+
+oc delete kfdef odh-modelmesh -n $MODELMESH_PROJECT --ignore-not-found=true &
```
Comment (bdattoma) [inline tests/Resources/Files/uninstall-model-serving.sh:6]: is `&` a mistake or desired?

Comment (lugi0) [reply tests/Resources/Files/uninstall-model-serving.sh:6]: it's desired, without it the delete command hangs and I cannot move on to the patch one

File: tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+${EXPECTED_INFERENCE_OUTPUT}=    {"model_name":"example-onnx-mnist__isvc-82e2bf7ea4","model_version":"1","outputs":[{"name":"Plus214_Output_0","datatype":"FP32","shape":[1,10],"data":[-8.233052,-7.749704,-3.4236808,12.363028,-12.079106,17.26659,-10.570972,0.7130786,3.3217115,1.3621225]}]}
+
+
+*** Test Cases ***
+Validate Model Serving quickstart
+    [Documentation]    Test the quickstart of the model serving repo, Temporary until included in RHODS
+    [Tags]    ModelMesh_Serving
+    # TODO: Replace with http://robotframework.org/robotframework/latest/libraries/Process.html#Run%20Process
+    Run    git clone ${MS_REPO}
+    Run    cd modelmesh-serving/quickstart && ./quickstart.sh ${ODH_NAMESPACE} ${MODEL_MESH_NAMESPACE}
```
Comment (bdattoma) [inline tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot:20]: can we check a particular output string/code to assess if this is a PASS or FAIL?

File: tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    [Documentation]    Test the inference result
+    [Tags]    ModelMesh_Serving
+    # make sure model is being served
+    # TODO: find better way to understand when model is being served
+    # One option is Triton pods being both 5/5 Ready
+    # Sleep  1m
+    ${MS_ROUTE} =    Run    oc get routes -n ${MODEL_MESH_NAMESPACE} example-onnx-mnist --template={{.spec.host}}{{.spec.path}}
+    ${AUTH_TOKEN} =    Run    oc sa new-token user-one -n ${MODEL_MESH_NAMESPACE}
+    ${inference_output} =    Run    curl -ks https://${MS_ROUTE}/infer -d @modelmesh-serving/quickstart/input.json -H "Authorization: Bearer ${AUTH_TOKEN}"
+    Should Be Equal As Strings    ${inference_output}    ${expected_inference_output}
```
Comment (bdattoma) [inline tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot:46]: `EXPECTED_INFERENCE_OUTPUT`

File: tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot (hunk 2 above)
Comment (bdattoma) [reply tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot:20]: ok let's say it is more a suite setup maybe. Okay anyway

**Review by bdattoma — DISMISSED** [review]

File: tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot (hunk 3 above)
Comment (lugi0) [reply tests/Tests/400__ods_dashboard/420__model_serving/420__model_serving.robot]: Done

**Review by jgarciao — APPROVED** [review]

**Review by pablofelix — APPROVED** [review]
lgtm

**Pull request merged**